gateway/src/apicast/policy/upstream/upstream.lua

```diff
@@ -58,6 +58,10 @@ function _M:rewrite(context)
   end
 end
 
+function _M:access(context)
+  context.route_upstream = context[self] or context.route_upstream
```

Comment:
is this ok?

Reply:
so basically you're exposing the upstream URL defined in the the proxy_config during the access phase here right? Do we do this anywhere else? One quick test to check this doesn't break anything is to run the entire test suite locally.

Reply:
yeah we are doing it [here in the routing policy](https://github.com/3scale/APIcast/blob/2ae88d8581288c9782d6124d072a7bbd1e75e55c/gateway/src/apicast/policy/routing/routing.lua#L56). I thought it would be nice to have a default variable in the context where, should a policy modify the upstream, it would store the new value, so I used the same variable here. This way we would know where to look for an updated upstream, regardless of which policy changed it. The reason I ask is I am not sure this was intended to be used this way and maybe I should use a dedicated variable for the Upstream policy instead.
According to the tests it doesn't break anything at the moment :P